src/snowflake/snowpark/session.py

```diff
@@ -1980,41 +1997,45 @@ def query_history(self) -> QueryHistory:
-        # implementation based upon: https://docs.snowflake.com/en/sql-reference/name-resolution.html
-        validate_object_name(table_name)
-        # note: object name could have dots, e.g, a table could be created via: create table "abc.abc" (id int)
-        # currently validate_object_name does not allow it, but if in the future we want to support the case, we need to
-        # update the implementation accordingly in this method
-        qualified_table_name = table_name.split(".")
-        if len(qualified_table_name) == 1:
-            # name in the form of "table"
-            tables = self._run_query(f"show tables like '{table_name}'")
-        elif len(qualified_table_name) == 2:
-            # name in the form of "schema.table" omitting database
-            # schema: qualified_table_name[0]
-            # table: qualified_table_name[1]
-            tables = self._run_query(
-                f"show tables like '{qualified_table_name[1]}' in schema {qualified_table_name[0]}"
-            )
-        elif len(qualified_table_name) == 3:
-            # name in the form of "database.schema.table"
-            # database: qualified_table_name[0]
-            # schema: qualified_table_name[1]
-            # table: qualified_table_name[2]
-            condition = (
-                f"database {qualified_table_name[0]}"
-                if qualified_table_name[1] == ""
-                else f"schema {qualified_table_name[0]}.{qualified_table_name[1]}"
-            )
+    def _table_exists(self, raw_table_name: Union[str, Iterable[str]]):
+        """ """
+        if isinstance(raw_table_name, str):
             tables = self._run_query(
-                f"show tables like '{qualified_table_name[2]}' in {condition}"
+                f"show tables like '{_strip_double_quotes_in_like_statement_in_table_name(raw_table_name)}'"
             )
         else:
-            # we do not support len(qualified_table_name) > 3 for now
-            raise SnowparkClientExceptionMessages.GENERAL_INVALID_OBJECT_NAME(
-                table_name
-            )
+            # implementation based upon: https://docs.snowflake.com/en/sql-reference/name-resolution.html
+            qualified_table_name = list(raw_table_name)
+            if len(qualified_table_name) == 1:
+                # name in the form of "table"
+                tables = self._run_query(
+                    f"show tables like '{_strip_double_quotes_in_like_statement_in_table_name(qualified_table_name[0])}'"
+                )
+            elif len(qualified_table_name) == 2:
+                # name in the form of "schema.table" omitting database
+                # schema: qualified_table_name[0]
+                # table: qualified_table_name[1]
+                tables = self._run_query(
+                    f"show tables like '{_strip_double_quotes_in_like_statement_in_table_name(qualified_table_name[1])}' in schema {qualified_table_name[0]}"
```

Comment:
Database and schema names are also identifiers. Do we need to do similar treatment as table?

Reply:
nope, `SHOW` command works like:
> SHOW tables like '<table_name>' in \<db\>.\<schema\>

<table_name> is expected to be a string constant as wrapped in single quotes `'`.

but db and schema after `in` can keep the double quotes as they are idenfitiers